### PR TITLE
[Do Not Merge] Kusto: Add support for attached databases

### PR DESCRIPTION
**Pull request by radennis**
### Latest improvements:
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i><br>
### Contribution checklist:
- [ ] I have reviewed the [documentation](https://github.com/Azure/azure-rest-api-specs#basics) for the workflow.
- [ ] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.
- [ ] The [OpenAPI Hub](https://aka.ms/openapiportal) was used for checking validation status and next steps.
### ARM API Review Checklist
- [ ] Service team MUST add the "WaitForARMFeedback" label if the management plane API changes fall into one of the below categories. 
- adding/removing APIs.
- adding/removing properties.
- adding/removing API-version. 
- adding a new service in Azure.

Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs.
- [ ] If you are blocked on ARM review and want to get the PR merged urgently, please get the ARM oncall for reviews (RP Manifest Approvers team under Azure Resource Manager service) from IcM and reach out to them. 
Please follow the link to find more details on [API review process](https://armwiki.azurewebsites.net/rp_onboarding/ResourceProviderOnboardingAPIRevieworkflow.html).

### CODE REVIEW

**Comment by adxsdknet** [comment]
# Automation for azure-sdk-for-net 
A PR has been created for you: 
https://github.com/azure/azure-sdk-for-net/pull/6253 
.NET SDK Commits: 
https://github.com/adxsdknet/azure-sdk-for-net/pull/6253/commits/552d689463b010402c18f7a4b6b574f418836464

File: specification/azure-kusto/resource-manager/Microsoft.Kusto/stable/2019-01-21/generated/AttachedDatabaseConfigurationsOperationsExtensions.cs
Comment (kpajdzik) [inline specification/azure-kusto/resource-manager/Microsoft.Kusto/stable/2019-01-21/generated/AttachedDatabaseConfigurationsOperationsExtensions.cs]: Is this checked-in by design?

Comment (radennis) [reply specification/azure-kusto/resource-manager/Microsoft.Kusto/stable/2019-01-21/generated/AttachedDatabaseConfigurationsOperationsExtensions.cs]: No sorry, deleted

**Comment by radennis** [comment]
@kpajdzik do you have any other comments or can approve?

**Review by kpajdzik — APPROVED** [review]

**Comment by kpajdzik** [comment]
> @kpajdzik do you have any other comments or can approve?

Could you resolve the merge conflicts? I'll merge after the resolution.

**Comment by radennis** [comment]
Thanks @kpajdzik. Please do not merge yet. The CR is here to make sure it is approved by the SDK and ARM folks before going forward. Thanks!

**Comment by kpajdzik** [comment]
> Thanks @kpajdzik. Please do not merge yet. The CR is here to make sure it is approved by the SDK and ARM folks before going forward. Thanks!

I'm adding ARM team if you need their review. I believe I'm one of the "SDK folks" so we have this checked :)

**Comment by radennis** [comment]
That's great, thanks @kpajdzik. Regarding ARM, would appreciate @KrisBash's approval since he attended the design review.

**Comment by sanjaiganesh** [comment]
KrishBash is ARM reviewer next week. I would defer to Krish this as Ravit suggested.

**Comment by radennis** [comment]
@KrisBash could you please take a look and approve per our design? Just to make it clear, we're not looking into pushing this change now. Just getting your approval, so that once the feature is complete we can add it to the swagger.

File: specification/azure-kusto/resource-manager/Microsoft.Kusto/stable/2019-01-21/kusto.json
Comment (KrisBash) [inline specification/azure-kusto/resource-manager/Microsoft.Kusto/stable/2019-01-21/kusto.json]: The property name here is plural, but the request object only takes a single clusterId + databaseName tuple. Should this take an array of tuples?

File: specification/azure-kusto/resource-manager/Microsoft.Kusto/stable/2019-01-21/kusto.json
Comment (KrisBash) [inline specification/azure-kusto/resource-manager/Microsoft.Kusto/stable/2019-01-21/kusto.json]: Is this description correct - EngagementFabric?  I haven't seen that phrase in this spec.

**Review by KrisBash — CHANGES_REQUESTED** [review]
Thanks for working closely with us on this one. - it looks as expected. I had a few questions in my comments.

File: specification/azure-kusto/resource-manager/Microsoft.Kusto/stable/2019-01-21/kusto.json
Comment (radennis) [reply specification/azure-kusto/resource-manager/Microsoft.Kusto/stable/2019-01-21/kusto.json]: We not supporting an array, because we thought it would be confusing for a user to have to provide all the list of databases each time he wants to add/remove a database and it could be risky if he omits a database and it would be removed mistakely from the list.

Yes, multiple databases for the same database can be created usingmultiple attachedDatabaseConfigurations. Whenenver just a single database is provided, the readonly list of attached databaseNames would return just the single database.

File: specification/azure-kusto/resource-manager/Microsoft.Kusto/stable/2019-01-21/kusto.json
Comment (radennis) [reply specification/azure-kusto/resource-manager/Microsoft.Kusto/stable/2019-01-21/kusto.json]: You're right I'll rename the parameter to singular (I rather keep the API simple)

File: specification/azure-kusto/resource-manager/Microsoft.Kusto/stable/2019-01-21/kusto.json
Comment (radennis) [reply specification/azure-kusto/resource-manager/Microsoft.Kusto/stable/2019-01-21/kusto.json]: No, I'll fix it, thanks

File: specification/azure-kusto/resource-manager/Microsoft.Kusto/stable/2019-01-21/kusto.json
Comment (KrisBash) [reply specification/azure-kusto/resource-manager/Microsoft.Kusto/stable/2019-01-21/kusto.json]: ok, thanks

**Review by KrisBash — APPROVED** [review]
LGTM

**Comment by radennis** [comment]
Thanks @KrisBash !

**Pull request closed without merging**
